### PR TITLE
Report the version of Xamarin.Mac that's used at runtime by the IDE

### PR DESCRIPTION
**Pull request by Therzok**
Fixes VSTS #577285 Report version of Xamarin.Mac that's used at runtime

### CODE REVIEW

File: main/external/Xamarin.Mac.version [hunk 1]
```diff
@@ -0,0 +1 @@
+4.4.0.36
```
Comment (slluis) [inline main/external/Xamarin.Mac.version]: What is this for?

File: main/src/addins/MacPlatform/MacPlatform.cs [hunk 2]
```diff
@@ -156,8 +156,42 @@ public MacPlatformService ()
...
+			return line.Split (':') [1].Trim ();
+		}
+
+		static Lazy<string> XamMacBuildInfo = new Lazy<string> (() => {
+			const string buildInfoResource = "Xamarin.Mac.buildinfo";
+			var asm = System.Reflection.Assembly.GetExecutingAssembly ();
+
+			string version, hash, branch;
+
+			using (var stream = asm.GetManifestResourceStream (buildInfoResource))
```
Comment (slluis) [inline main/src/addins/MacPlatform/MacPlatform.cs]: This method should be better protected against crashes, that may happen if the format of buildinfo changes for some reason.

File: main/external/Xamarin.Mac.version (hunk 1 above)
Comment (Therzok) [reply main/external/Xamarin.Mac.version]: Remnant.

File: main/src/addins/MacPlatform/MacPlatform.cs (hunk 2 above)
Comment (Therzok) [reply main/src/addins/MacPlatform/MacPlatform.cs]: 👍

**Comment by Therzok** [comment]
Fixed

**Pull request merged**